pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -515,7 +526,7 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 
 		if *vm.ProvisioningState == "Failed" {
 			// If VM failed provisioning, delete it so it can be recreated
-			err = s.virtualMachinesSvc.Delete(ctx, vmSpec)
+			err = s.Delete(ctx)
```

Comment:
am I right to say that the commit message mentions disk but this PR will also introduce a new network interface deletion?

Reply:
Thats correct, let me update PR message and commit message

Reply:
Will the deleted network interface get recreated again?

Reply:
Yep! its part of create workflow, since Exists returns false, cluster api would call Create again https://github.com/openshift/cluster-api-provider-azure/blob/106e77fea15a9a0a39776b3edbc5666aa337ca70/pkg/cloud/azure/actuators/machine/reconciler.go#L84


The upstream PR is https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/e1f358dc8cd5675911178ab180d04e613aec9e18